modules/perSystem.nix

```diff
@@ -35,17 +39,41 @@ in
         };
       });
       apply = modules: system:
-        (lib.evalModules {
-          inherit modules;
-          prefix = [ "perSystem" system ];
-          specialArgs = {
-            inherit system;
-          };
-        }).config;
+        let
+          inherit (lib.evalModules {
+            inherit modules;
+            prefix = [ "perSystem" system ];
+            specialArgs = {
+              inherit system;
+            };
+          }) options;
+        # Almost like `config`, except undefined options are filtered away.
+        # This definition is incomplete, but is good enough for our purposes.
+        # Limitations:
+        # - Doesn't handle undefined options in submodules.
+        # - Doesn't handle undefined options that have a prefix, e.g. `_module.args`.
+        # TODO: Upstream a good version of this to Nixpkgs, probably as an extra result
+        # from `evalModules`, e.g. `result.configFiltered`.
```

Comment:
We should be careful about adding attributes. They're not like methods that have no per-object overhead.